core/src/main/java/org/bitcoinj/core/Wallet.java

```diff
@@ -3971,7 +3971,8 @@ public boolean canSignFor(Script script) {
     protected List<UTXO> getStoredOutputsFromUTXOProvider() throws UTXOProviderException {
         UTXOProvider utxoProvider = checkNotNull(vUTXOProvider, "No UTXO provider has been set");
         List<UTXO> candidates = new ArrayList<UTXO>();
-        List<DeterministicKey> keys = getActiveKeyChain().getLeafKeys();
+        List<ECKey> keys = getImportedKeys();
+        keys.addAll(getActiveKeyChain().getLeafKeys());
```

Comment:
Yeah, we're copying it deliberately and yes we could document that. In general we should aim for owned-by-caller by default, and if we return a pointer to mutable internal state for performance reasons or otherwise it should be that which is explicitly documented and called out.